org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/XtextDocumentProvider.java

```diff
@@ -410,16 +411,29 @@ public long getModificationStamp(Object element) {
 	
 	@Override
 	public boolean isSynchronized(Object element) {
-		ElementInfo info = getElementInfo(element);
-		long synchronizationStamp;
-		if (info instanceof FileInfo) {
-			synchronizationStamp = ((FileInfo) info).fModificationStamp;
-		} else if (info instanceof URIInfo) {
-			synchronizationStamp = ((URIInfo) info).synchronizationStamp;
-		} else {
-			return super.isSynchronized(element);
+		if (element instanceof IFileEditorInput) {
```

Comment:
I personally avoid many nested if levels, so I would revert the condition and also use modern instanceof
if (!(element instanceof IFileEditorInput input)){
   return super.isSynchronized(element);
}

// continue with the new code
// "input" is already defined here with the right type

Reply:
Currently the project uses java 11. Am I allowed to upgrade to java 17 or 21?

Reply:
> Am I allowed to upgrade to java 17 or 21?

I would say, definitely not **in the context of this ticket**, but it is a good question and probably deserves a dedicated ticket in Xtext.

Note that Eclipse Platform is switched to Java 17 in Eclipse 4.28, including ecj (see https://github.com/eclipse-jdt/eclipse.jdt.core/issues/886).

Reply:
there is already a ticket for drop 11 17 minimum
https://github.com/eclipse/xtext/issues/2686